doc/stdenv/platform-support-tiers.chapter.md

```diff
@@ -0,0 +1,87 @@
+# Platform Support Tiers {#platform-support-tiers}
+
+In the context of Nixpkgs, a *platform* can mean a CPU architecture, a
+[cross-compilation](#chap-cross) target or a C library option.
+
+Nixpkgs defines 7 platform *support tiers*, meaning levels of support for each platform.
+
+**TODO**: change build successfully to a definition of "work".
+
+## Tier 1 {#platform-support-tier-1}
+
+These platforms receive the highest level of support, meaning that every
+platform-compatible package in Nixpkgs must successfully build in the Hydra and OfBorg CIs.
+**TODO**: this seems to not be entirely true for Hydra, but true for OfBorg.
+
+- `x86_64-linux`, `gcc` + `glibc`
+
+## Tier 2 {#platform-support-tier-2}
+
+Many of the packages are required to build successfully on these platforms in CI, the rest are supported on a best-effort basis by dedicated platform maintainers.
+**TODO**: link teams
+
+- `aarch64-linux`, `gcc+glibc`
+- `x86_64-darwin`, `clang+Darwin/macOS`
+
+## Tier 3 {#platform-support-tier-3}
+
+None of the packages for these platforms are required to build successfully in
+CI, but it is expected that most of them will build successfully.
+
+- `i686-linux`, `gcc`+`glibc`
+- `armv{6,7,8}*-linux`, `gcc`+`glibc`
+- `armv{6,7,8}*-linux`, `gcc`+`glibc`, cross-compilation
+- `aarch64-linux`, `gcc`+`glibc`, cross-compilation
+- `mipsel-linux`, `gcc`+`glibc`
+- `x86_64-linux`, `gcc`+`musl`
+
+## Tier 4 {#platform-support-tier-4}
+
+Some of the packages for these platforms are expected to work. The
+[`exotic-platform-maintainers`
+Team](https://github.com/orgs/NixOS/teams/exotic-platform-maintainers) is
+responsible for this platform.
+
+- `aarch64-none`
+- `avr`
+- `arm-none`
+- `i686-none`
+- `x86_64-none`
+- `powerpc-none`
+- `powerpcle-none`
+- `x86_64-mingw32`
+- `i686-mingw32`
+- `x86_64-linux`, `gcc`+`musl` — static
+- `x86_64-linux`, `clang`+`glibc`
+- `x86_64-linux`, `clang`+`glibc` — `llvm` linker
+- `x86_64-linux` — Android
+- `aarch64-linux` — Android
+- `armv{7,8}-linux` — Android
+
+## Tier 5 {#platform-support-tier-5}
+
+A small number of packages might build successfully on these platforms.
+
+- `x86_64-linux`, `gcc`+`glibc` — static
+- `x86_64-linux`, `gcc`+`glibc` — `llvm` linker
+
+## Tier 6 {#platform-support-tier-6}
+
+These platforms are defined in Nixpkgs, but no packages are expected to build on them.
+
+- `wasm-wasi`
+- `powerpc64le-linux`, `gcc`+`glibc`
+
+## Tier 7 {#platform-support-tier-7}
+
+No current support, but previous support or clear path to add support.
+
+- `aarch64-darwin`
```

Comment:
This seems to be false now, since OfBorg builds for this platform.

Reply:
Yes, it looks like right now after all the work done, two macOS versions are mostly on par. And upstream pretends that the latter is the continuation of the former in some sense.

Reply:
> I'm inclined towards leaving this at the state it was when the RFC was accepted, since I don't have the resources to update the list on my own.

In that case you should probably close this PR.

Adding inaccurate data to the documentation is not helpful.

Reply:
Well, it's inaccurate until someone fixes it, and seeing how nothing has happened in 3 years, this PR might at least get some things moving.

Reply:
But let's see if we gather more up to date info with this PR. Thank you for your contributions to that end.